### ISSUE TITLE
Drop Last Trick

### ISSUE THREAD

**Issue by onderakacik** [zhouziyu02/SDformer#4]
https://github.com/zhouziyu02/SDformer/blob/eb1baa7e6513d185fbf993960b6851667b6526d5/data_provider/data_factory.py#L29

Hi,
I really enjoyed reading your paper. However, I noticed in the data factory script drop_last is set True for the test data loader. This could cause dropping up to "batch_size - 1" points during testing. Thus, it can lead to unfair comparisons with models that evaluate on the whole test set. I assume this error propagated from Time-Series-Library. However, they set this to "False" after [this paper](https://www.vldb.org/pvldb/vol17/p2363-hu.pdf) came out (see figure 4 from the paper).
I hope this helps to improve the quality of your work.
Best,
Önder

**Comment by zhouziyu02** [comment]
Thanks for your interest in our research. I really appreciate that you point out the "drop_last" setting, and your advice is helpful for improving the quality of our paper. Actually, about 1.5 years ago when I was conducting the SDformer experiment, the "drop_last" issue—originating from the implementation of Informer, an extremely famous paper according to FITS (ICLR2024)—had not yet been proposed or discussed within the community. I will absolutely modify it in our training setting.
